tests/views/test_message.py

```diff
@@ -415,20 +417,29 @@ def test_get_thread_success(self, mock_request, mock_get_jwt):
         self.assertIn("MISC".encode(), response.data)
 
     @requests_mock.mock()
-    @patch("response_operations_ui.controllers.message_controllers._get_jwt")
-    def test_get_thread_with_deleted_user_cannot_be_replied_to(self, mock_request, mock_get_jwt):
-        sign_in_with_permission(self, mock_request, user_permission_messages_edit_json)
-        mock_get_jwt.return_value = "blah"
-        with open(f"{project_root}/test_data/message/thread_missing_respondent.json") as thread_json:
-            missing_user_json = json.load(thread_json)
-        with self.client.session_transaction() as session:
-            session["user_id"] = "test-id"
-        mock_request.get(url_get_thread, json=missing_user_json)
-        mock_request.put(url_update_label)
-        mock_request.get(url_get_surveys_list, json=survey_list)
-        self.before()
-        response = self.client.get("/messages/threads/fb0e79bd-e132-4f4f-a7fd-5e8c6b41b9af")
-        self.assertIn("Cannot reply or modify conversation as the respondent has been deleted".encode(), response.data)
+    def test_open_thread_with_deleted_respondent(self, mock_request):
+        response = self._get_thread_with_deleted_respondent(mock_request, missing_user_json)
+        self.assertIn("You can no longer send a message as this respondent has been deleted".encode(), response.data)
+        self.assertIn("Close conversation".encode(), response.data)
+
+    @requests_mock.mock()
+    def test_closed_thread_with_deleted_respondent(self, mock_request):
+        missing_user_json_closed_thread = missing_user_json.copy()
```

Comment:
Does this work as expected? If you make a copy then modify it, doesn't the original get modified too?  I've been using deepcopy to make sure they're completely brand new, but if this works in the same way then great.

Reply:
Worth checking

Reply:
Copy is fine https://docs.python.org/3/library/copy.html as there isn't any compound objects I don't need to copy them as well, which would be deepcopy